doc/tutorials/core/basic_geometric_drawing/basic_geometric_drawing.markdown

```diff
@@ -120,95 +77,31 @@ Explanation
         -   The line thickness is set to **thickness** (in this case 2)
         -   The line is a 8-connected one (**lineType** = 8)
     -   *MyEllipse*
-        @code{.cpp}
-        void MyEllipse( Mat img, double angle )
-        {
-            int thickness = 2;
-            int lineType = 8;
-
-            ellipse( img,
-               Point( w/2.0, w/2.0 ),
-               Size( w/4.0, w/16.0 ),
-               angle,
-               0,
-               360,
-               Scalar( 255, 0, 0 ),
-               thickness,
-               lineType );
-        }
-        @endcode
+        @snippet cpp/tutorial_code/core/Matrix/Drawing_1.cpp myellipse
+
         From the code above, we can observe that the function @ref cv::ellipse draws an ellipse such
         that:
 
         -   The ellipse is displayed in the image **img**
-        -   The ellipse center is located in the point **(w/2.0, w/2.0)** and is enclosed in a box
-            of size **(w/4.0, w/16.0)**
+        -   The ellipse center is located in the point **(w/2, w/2)** and is enclosed in a box
+            of size **(w/4, w/16)**
```

Comment:
Actually you should not remove the .0 parts here. Ellipse centers are defined as Point2f, resulting in sub pixel accuracy for the location of the circle center. The .0 ensures that w, which is an integer, is divided by a float to ensure that nothing gets cut off due to integer division.